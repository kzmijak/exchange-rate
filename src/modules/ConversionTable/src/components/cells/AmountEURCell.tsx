import { FC } from "react";
import { ConversionTableRow } from "../../models/ConversionTableRow";
import { Box, Typography } from "@mui/joy";
import { convertPlnToEur } from "utils/convertCurrency";

type AmountEURCellProps = { row: ConversionTableRow };

export const AmountEURCell: FC<AmountEURCellProps> = ({ row }) => {
  return (
    <Box component="td">
      <Typography textAlign="right">
        {convertPlnToEur(row.amountPLN)}
      </Typography>
    </Box>
  );
};
