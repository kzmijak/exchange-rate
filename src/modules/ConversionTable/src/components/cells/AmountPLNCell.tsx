import { FC } from "react";
import { Box, Typography } from "@mui/joy";
import { ConversionTableRow } from "../../models/ConversionTableRow";

type AmountPLNCellProps = { row: ConversionTableRow };

export const AmountPLNCell: FC<AmountPLNCellProps> = ({ row }) => {
  return (
    <Box component="td">
      <Typography textAlign="right">{row.amountPLN}</Typography>
    </Box>
  );
};
