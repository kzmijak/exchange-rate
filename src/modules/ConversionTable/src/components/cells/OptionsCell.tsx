import { Box, Button } from "@mui/joy";
import { FC } from "react";
import { ConversionTableRow } from "../../models/ConversionTableRow";

type OptionsCellProps = {
  row: ConversionTableRow;
};

export const OptionsCell: FC<OptionsCellProps> = ({ row }) => {
  const handleClick = () =>
    alert(`I REALLY WANT TO DELETE THIS WHOLE GUY ${row.title} `);
  return (
    <Box component="td">
      <Button onClick={handleClick}>Delete</Button>
    </Box>
  );
};
