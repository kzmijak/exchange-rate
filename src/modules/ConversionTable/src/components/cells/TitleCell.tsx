import { FC } from "react";
import { Box, Typography } from "@mui/joy";
import { ConversionTableRow } from "../../models/ConversionTableRow";

type TitleCellProps = { row: ConversionTableRow };

export const TitleCell: FC<TitleCellProps> = ({ row }) => {
  return (
    <Box component="td">
      <Typography>{row.title}</Typography>
    </Box>
  );
};
