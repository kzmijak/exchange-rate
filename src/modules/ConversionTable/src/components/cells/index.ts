export { TitleCell } from "./TitleCell";
export { AmountEURCell } from "./AmountEURCell";
export { AmountPLNCell } from "./AmountPLNCell";
export { OptionsCell } from "./OptionsCell";
