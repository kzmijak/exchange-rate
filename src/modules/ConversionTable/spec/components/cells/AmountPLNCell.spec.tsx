import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import { AmountPLNCell } from "../../../src/components/cells/AmountPLNCell";
import { row1 } from "../../__fixtures__/rows";

describe("AmountPLNCell", () => {
  const Component = () => <AmountPLNCell row={row1} />;

  it("should match snapshot", () => {
    expect(render(<Component />).container).toMatchSnapshot();
  });

  it("should show amount in zlotys", () => {
    render(<Component />);
    expect(screen.getByText("100")).toBeInTheDocument();
  });
});
