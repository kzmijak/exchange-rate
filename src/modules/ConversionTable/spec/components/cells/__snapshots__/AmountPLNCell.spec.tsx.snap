// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AmountPLNCell > should match snapshot 1`] = `
<div>
  <td
    class="MuiBox-root css-0"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-1dh3lps-JoyTypography-root"
    >
      100
    </p>
  </td>
</div>
`;
